### ISSUE TITLE
Wrong output tensor shape from experimental.all_gather_async

### ISSUE THREAD

**Issue by hkwonTT** [tenstorrent/tt-metal#24399]
### Describe the bug

The tensor ttnn.experimental.all_gather_async generates has wrong shape.

It looks like ttnn.experimental.all_gather_async doesn't do unpad at the end.

### Steps to reproduce the issue

```
import ttnn
import torch
def test_all_gather_async(shape, gather_dim):
    ttnn.set_fabric_config(ttnn.FabricConfig.FABRIC_1D)
    mesh_device = ttnn.open_mesh_device(ttnn.MeshShape(1, 8))
    try:
        input = torch.rand(shape, dtype=torch.float32)

        sharded_input = ttnn.from_torch(input, layout=ttnn.TILE_LAYOUT, device=mesh_device, mesh_mapper=ttnn.ShardTensorToMesh(mesh_device, dim=1))
        print("input shape : ", sharded_input.shape)

        compute_grid_size = mesh_device.compute_with_storage_grid_size()
        ccl_sub_device_crs = ttnn.CoreRangeSet(
            {ttnn.CoreRange(ttnn.CoreCoord(0, 0), ttnn.CoreCoord(compute_grid_size.x - 1, compute_grid_size.y - 1))}
        )
        semaphore = ttnn.create_global_semaphore(mesh_device, ccl_sub_device_crs, 0)

        sharded_output_async = ttnn.experimental.all_gather_async(sharded_input, dim=gather_dim, 
        cluster_axis=1, mesh_device=mesh_device, topology=ttnn.Topology.Linear, multi_device_global_semaphore = semaphore)
        print("output shape : ", sharded_output_async.shape)

        output = ttnn.to_torch(sharded_output_async, device=mesh_device, mesh_composer=ttnn.ConcatMeshToTensor(mesh_device, dim=1))

    except Exception as e:
        print(f"An error occurred: {e}")
    finally:
        ttnn.close_mesh_device(mesh_device)
        ttnn.set_fabric_config(ttnn.FabricConfig.DISABLED)

test_all_gather_async((1, 8, 64, 64), 0)
test_all_gather_async((1, 8, 64, 63), 0)
test_all_gather_async((1, 8, 63, 64), 0)
test_all_gather_async((1, 8, 63, 1), 0)
```

output:
```
input shape :  Shape([1, 1, 64, 64])
output shape :  Shape([8, 1, 64, 64])

input shape :  Shape([1, 1, 64, 63])
output shape :  Shape([8, 1, 64, 64])

input shape :  Shape([1, 1, 63, 64])
output shape :  Shape([8, 1, 64, 64])

input shape :  Shape([1, 1, 63, 1])
output shape :  Shape([8, 1, 64, 32])
```

### Expected behavior

```
input shape :  Shape([1, 1, 64, 64])
output shape :  Shape([8, 1, 64, 64])

input shape :  Shape([1, 1, 64, 63])
output shape :  Shape([8, 1, 64, 63])

input shape :  Shape([1, 1, 63, 64])
output shape :  Shape([8, 1, 63, 64])

input shape :  Shape([1, 1, 63, 1])
output shape :  Shape([8, 1, 63, 1])
```

**Comment by ntarafdar** [comment]
@grechsteinerTT , can you tack this on as part of your AG fixes for RM.

**Comment by grechsteinerTT** [comment]
@hkwonTT these shapes are now supported when gathering on dim 0.

FYI the API for the op has changed slightly, you can reference this test file for how to use the updated API: https://github.com/tenstorrent/tt-metal/blob/main/tests/nightly/t3000/ccl/test_minimal_all_gather_async.py
